src/memote/Memote.jl

```diff
@@ -0,0 +1,289 @@
+"""
+# Memote
+
+Main entry point into metabolic model quality tests.
+"""
+module Memote
+
+using DocStringExtensions
+using COBREXA
+using Test
+using JuMP
+using ..Utils
+
+include(joinpath("Config.jl"))
+include(joinpath("checks", "Annotation.jl"))
+include(joinpath("checks", "Basic.jl"))
+include(joinpath("checks", "Biomass.jl"))
+include(joinpath("checks", "Consistency.jl"))
+include(joinpath("checks", "Energy.jl"))
+include(joinpath("checks", "GPRAssociation.jl"))
+include(joinpath("checks", "Metabolite.jl"))
+include(joinpath("checks", "Network.jl"))
+include(joinpath("checks", "Reaction.jl"))
+
+using .Config
+using .Annotation
+using .Basic
+using .Biomass
+using .Consistency
+using .Energy
+using .GPRAssociation
+using .Metabolite
+using .Network
+using .Reaction
+
+"""
+$(TYPEDSIGNATURES)
+
+Run the standard memote test suite on `model` using `optimizer` to solve
+optimization problems and configure the test parameters with `config`.
+
+Note, for best results, make sure the model can be converted into a
+`COBREXA.StandardModel`.
+"""
+function run_tests(model, optimizer; config = Config.memote_config)
+    @testset "Metabolic model tests" begin
+
+        @testset "Basic information" begin
+            @test Basic.model_has_name(model)
+            @test Basic.model_has_reactions(model)
+            @test Basic.model_has_metabolites(model)
+            @test Basic.model_has_genes(model)
+            @test Basic.model_metabolic_coverage_exceeds_minimum(model; config)
+            @test Basic.model_has_compartments(model)
+            @test Basic.model_solves_in_default_medium(model, optimizer; config)
+        end
+
+        @testset "Annotations" begin
+            @testset "Reaction annotations" begin
+                @test isempty(Annotation.find_all_unannotated_reactions(model))
+
+                frac_annotated =
+                    1 -
+                    reduce(
+                        min,
+                        length.(values(Annotation.find_database_unannotated_reactions(model; config))),
+                    ) / n_reactions(model)
```

Comment:
do not delete by zero (unnecessary exception risk), instead test as
```
@test n_annotated >= n_reactions(model) * minimum_frac_annotated
```
(note the `>=`! the original test will always fail if requesting 100% coverage, even with 100% coverage)

Reply:
(this might belong to a different issue tho)